app/controllers/payment_controller.rb

```diff
@@ -5,15 +5,21 @@ def available_refunds
     if current_user
       attendee_id = params.require(:attendee_id)
       competition_id, user_id = attendee_id.split("-")
+      competition = Competition.find(competition_id)
 
-      ms_registration = MicroserviceRegistration.includes(:competition, :payment_intents)
-                                                .find_by(competition_id: competition_id, user_id: user_id)
-      return render status: :bad_request, json: { error: "Registration not found" } unless ms_registration.present?
+      registration = if competition.registration_version_v3?
+                       Registration.includes(:payment_intents)
+                                   .find_by(competition_id: competition_id, user_id: user_id)
+                     else
+                       MicroserviceRegistration.includes(:payment_intents)
+                                               .find_by(competition_id: competition_id, user_id: user_id)
```

Comment:
Can leverage https://github.com/thewca/worldcubeassociation.org/pull/10133 once that's merged